### PR TITLE
[numpy] add decimals argument to round

### LINKED ISSUE #65908: Function request: np.around (alias with torch.round)
Add `around` as an alias for `round` and extend `round` to handle the "decimals" argument like NumPy: https://numpy.org/doc/stable/reference/generated/numpy.around.html?highlight=around#numpy.around. 

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @mruberry @rgommers

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes #65908 

Added a new overload instead of updating the current signature. (Had issues with JIT and **maybe** it would have been FC breaking)

TODO:

* [x] Don't compute `std::pow(10, decimals)` for each element.
* [x] Update docs (https://docs-preview.pytorch.org/66195/generated/torch.round.html?highlight=round#torch.round)
* [x] Add tests
* ~~Should we try to make it composite?~~
* ~~Should we add specialized test with more values of `decimals` outside of OpInfo with larger range of values in input tensor?~~

cc @mruberry @rgommers

### CODE REVIEW

**Review by mruberry — COMMENTED** [review]
Hey @kshitij12345!

Overall this looks really good. The doc, signature, and testing updates are very precise.

My question is how numerically reasonable is this kernel? Multiplying and dividing floats by powers of ten is worrying, especially for smaller float types like float16. I'm surprised the larger reference numerics test pass for those small floats given that NumPy's implementation seems to be casting to double. 

What are your thoughts on the numerics? Is rounding with a decimal value a challenge for small float types? Should we consider an alternate algorithm? For example, we could subtract the integer part of the number before working with the decimal.

Like... possibly alternate implementation might involve rounding and then adding back the decimals using the frac function?

**Comment by kshitij12345** [comment]
@mruberry gentle ping :)

**Review by mruberry — APPROVED** [review]
Thanks for the pings, @kshitij12345!

This looks great! I made a few small doc tweaks if you'd like to take a look.

**Comment by kshitij12345** [comment]
@mruberry any update?

**Comment by kshitij12345** [comment]
@mruberry gentle ping :)

**Comment by cpuhrsch** [comment]
Hello @kshitij12345 - could you do a rebase please?

File: aten/src/ATen/native/cuda/UnaryFractionKernels.cu [hunk 1]
```diff
@@ -117,17 +117,6 @@ __host__ __device__ static inline double nearbyint_wrapper(double a) {
   return ::nearbyint(a);
 }
 
-__host__ __device__ static inline c10::complex<float> nearbyint_wrapper(c10::complex<float> a) {
```
Comment (mruberry) [inline aten/src/ATen/native/cuda/UnaryFractionKernels.cu:120]: I wonder if we need to add this back.

@cpuhrsch generously fixed many of the internal build issues, but clangtidy is complaining about a slew of intrinsic issues like

```
use of undeclared identifier '__builtin_ia32_reduce_fmul_pd512'; did you mean '__builtin_ia32_vpermilpd512'?
../../../../mnt/gvfs/third-party2/llvm-fb/6dee269d7fd86fcf635dc05968a63300f6c89fc7/12/platform009/d3a69bd/lib/clang/stable/include/avx512fintrin.h:9316:10: '__builtin_ia32_vpermilpd512' declared here
    return __builtin_ia32_reduce_mul_q512(__W);
           ^
```

Comment (kshitij12345) [reply aten/src/ATen/native/cuda/UnaryFractionKernels.cu:120]: Hmm. Will add it back.

Thanks @cpuhrsch!

**Comment by kshitij12345** [comment]
@mruberry please check the internal builds :) Thanks

**Comment by kshitij12345** [comment]
@mruberry does this look good now? Thanks!

**Comment by mruberry** [comment]
Hey @kshitij12345, sorry to make you wait. I'm clearing out old PRs now. Would you rebase this and ping me? I'll get it merged ASAP.

**Comment by kshitij12345** [comment]
@mruberry done :)

**Comment by kshitij12345** [comment]
@mruberry I think there is logical conflict or maybe I did something wrong with merge. Will fix it.

**Comment by mruberry** [comment]
> @mruberry I think there is logical conflict or maybe I did something wrong with merge. Will fix it.

OK; just ping me when it's ready.

File: aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp [hunk 2]
```diff
@@ -231,6 +231,22 @@ CREATE_UNARY_UFUNC(tanh);
 CREATE_UNARY_UFUNC(trunc);
 CREATE_UNARY_UFUNC(conj_physical);
 
+// With addition of `round.decimals` overload, using CREATE_UNARY_UFUNC leads
+// to unresolved overload.
```
Comment (kshitij12345) [inline aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp:235]: Currently haven't added the `decimals` support for Sparse CSR
cc: @IvanYashchuk @krshrimali @mruberry

Comment (IvanYashchuk) [reply aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp:235]: Alternatively, the `CREATE_UNARY_UFUNC` macro could be modified to use `ATEN_FN` and `ATEN_FN2` to get unambiguous functions.

It's also fine to leave `round` separate as it is in this PR.

Comment (kshitij12345) [reply aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp:235]: Thanks @IvanYashchuk. Let's keep it seperate for now.

**Comment by kshitij12345** [comment]
Ping @mruberry. This should be ready now :)

**Pull request closed without merging**